src/pages/q/[[...all]].tsx

```diff
@@ -175,11 +193,27 @@ function BrandPage({serverState}: any) {
   )
 }
 
+const limiter = rateLimit({
+  interval: 60 * 1000,
+  uniqueTokenPerInterval: 500,
+})
+
 export const getServerSideProps: GetServerSideProps = async function ({
   req,
   query,
   res,
 }) {
+  try {
+    await limiter.check(res, 60, 'SEARCH_CACHE_TOKEN') // 60 requests per minute
+  } catch (error) {
+    res.setHeader('Retry-After', 60)
+    res.statusCode = 429
+
+    return {
+      props: {error: 'Rate limit exceeded, please try again in 60 seconds.'},
+    }
```

Comment:
if there is an error, we set status code to 429 and `Retry-After` header to 60 seconds which is the interval we have set.